pyvista/core/composite.py

```diff
@@ -690,3 +703,244 @@ def copy(self, deep=True):
         newobject.copy_meta_from(self, deep)
         newobject.wrap_nested()
         return newobject
+
+    def set_active_scalars(
+        self, name: Optional[str], preference: str = 'cell', allow_missing: bool = False
+    ) -> Tuple[FieldAssociation, np.ndarray]:  # type: ignore
+        """Find the scalars by name and appropriately set it as active.
+
+        To deactivate any active scalars, pass ``None`` as the ``name``.
+
+        Parameters
+        ----------
+        name : str or None
+            Name of the scalars array to assign as active.  If
+            ``None``, deactivates active scalars for both point and
+            cell data.
+
+        preference : str, optional
+            If there are two arrays of the same name associated with
+            points or cells, it will prioritize an array matching this
+            type.  Can be either ``'cell'`` or ``'point'``.
+
+        allow_missing : bool, optional
+            Allow missing scalars in part of the composite dataset. If all
+            blocks are missing the array, it will raise a ``KeyError``.
+
+        Returns
+        -------
+        pyvista.FieldAssociation
+            Field association of the scalars activated.
+
+        numpy.ndarray
+            An array from the dataset matching ``name``.
+
+        Notes
+        -----
+        The number of components of the data must match.
+
+        """
+        data_assoc: List[Tuple[FieldAssociation, np.ndarray, MultiBlock]] = []
+        for block in self:
+            if block is not None:
+                if isinstance(block, MultiBlock):
+                    field, scalars = block.set_active_scalars(
+                        name, preference, allow_missing=allow_missing
+                    )
+                else:
+                    try:
+                        field, scalars = block.set_active_scalars(name, preference)
+                    except KeyError as err:
+                        if not allow_missing:
+                            raise err
+                        block.set_active_scalars(None, preference)
+                        field, scalars = FieldAssociation.NONE, np.array([])
+
+                if field != FieldAssociation.NONE:
+                    data_assoc.append((field, scalars, block))
+
+        if name is None:
+            return FieldAssociation.NONE, np.array([])
```

Comment:
If my hunch is correct that scalars are normally `pyvista_ndarray`s then we should consider returning these for type consistency:
```suggestion
                        field, scalars = FieldAssociation.NONE, pyvista.pyvista_ndarray([])

                if field != FieldAssociation.NONE:
                    data_assoc.append((field, scalars, block))

        if name is None:
            return FieldAssociation.NONE, pyvista.pyvista_ndarray([])
```

Might need a `from` import to keep the line under 100 characters, haven't checked.

Reply:
Implemented in aee6ffa58ee0f26d183b464f761ef056f110cae2